### DISCUSSION TITLE
Creating Guest Sessions?

### DISCUSSION THREAD

**Discussion by dr-starchildren** [nextauthjs/next-auth#1237] [Help]
I'm looking to implement a guest session. The purpose is to allow an anonymous user to begin using the site (e.g. playing games) and store their progress in a database. They'll be prompted to create an account, but until they do I want them to be able play and save their progress. Once they create an account, I want to transfer the guest session to a user entry. 

Something like this in firebase:
https://firebase.google.com/docs/auth/web/anonymous-auth

Is this possible with next-auth? If so, how do you create the session without using the login page?

**Comment by luisrudge** [comment]
@dr-starchildren any luck on this? I'm also using a service that supports anonymous auth and I'm looking to build something to support it!

**Reply by dr-starchildren** [reply]
I haven't found a good solution to this outside of firebase. It would be a nice feature to have, but it's probably low on the list.
